### ISSUE TITLE
[telemetry] Web resource cache and bundle metrics in ph-oton-app

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#53]
## Goal

Instrument the web resource cache and resource bundling in `ph-oton-app`.

**Lowest priority of the ph-telemetry work** — this path is hot only during startup and the first requests after a deployment. File it, do the higher tiers first.

## Where

**1. `ph-oton-app/src/main/java/com/helger/photon/app/resource/WebSiteResourceCache.java`**

```java
public static WebSiteResource getOrCreateResource (@NonNull EWebSiteResourceType eResourceType, ...)
public static EChange removeFromCache (@NonNull EWebSiteResourceType eType, ...)
public static EChange clearCache ()
public static boolean isCacheEnabled ()
```

A hit/miss counter here answers whether the cache is doing anything, and whether someone left `setCacheEnabled (false)` on in production — a real and easy-to-miss misconfiguration.

**2. `ph-oton-app/src/main/java/com/helger/photon/app/resource/WebSiteResourceBundleManager.java`**

Bundle creation and the startup read loop (which logs `"Successfully read N resource bundles"` and skips stale bundles with `"Skipping resource bundle '...'"`). A counter of skipped/stale bundles is useful: stale bundles mean the on-disk bundle cache is being regenerated more often than expected.

**3. `ph-oton-app/src/main/java/com/helger/photon/app/resource/WebSiteResource.java`**

The constructor computes a content hash (message digest) over the resource — the actual I/O plus CPU cost. A histogram here is optional; only add it if (1) and (2) turn out not to be enough.

## What to emit

Extend `com.helger.photon.app.CAppTelemetry` (created by the UI-rendering issue; create it here if that one has not landed yet):

| Constant | Value |
|---|---|
| `METRIC_RESOURCE_CACHE_ACCESS` | `photon.resource.cache.access` |
| `METRIC_RESOURCE_BUNDLES_CREATED` | `photon.resource.bundles.created` |
| `METRIC_RESOURCE_BUNDLES_SKIPPED` | `photon.resource.bundles.skipped` |
| `ATTR_RESOURCE_CACHE_HIT` | `photon.resource.cache.hit` |
| `ATTR_RESOURCE_TYPE` | `photon.resource.type` (the `EWebSiteResourceType` — bounded: CSS / JS) |
| `UNIT_ACCESS` | `{access}` |
| `UNIT_BUNDLE` | `{bundle}` |

- **Counter** `photon.resource.cache.access`, attributes `hit` (`true`/`false`) + `type`.
- **Counter** `photon.resource.bundles.created`, attribute `type`.
- **Counter** `photon.resource.bundles.skipped`, attribute `type`.

No spans. This is startup-time work that does not belong to any request trace, and a span with no parent adds noise without adding information.

## Cardinality

`EWebSiteResourceType` is a two-value enum — safe.

The resource **path** is bounded in principle (the set of CSS/JS files shipped with the application) but can be in the hundreds and is the kind of dimension that quietly multiplies with everything else. Leave it off the metrics; if per-resource detail is ever needed, the log lines already carry it.

Content hashes must not be used as attributes at all.

## POM

`ph-oton-app/pom.xml` needs a new dependency if the UI-rendering issue has not already added it (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Cache hit and miss both counted, so a hit *ratio* is derivable.
- [ ] No resource path and no content hash on any attribute.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.6.0**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-app/.../app/CAppTelemetry.java` | extended - 3 metric names, 2 attribute keys, 2 units |
| `ph-oton-app/.../app/resource/WebSiteResourceMetrics.java` | new, public - 3 counters |
| `ph-oton-app/.../app/resource/WebSiteResourceTelemetry.java` | new, package-private emitter |
| `ph-oton-app/.../app/resource/WebSiteResourceCache.java` | modified |
| `ph-oton-app/.../app/resource/WebSiteResourceBundleManager.java` | modified |

No POM change - ph-oton-app already declares `ph-telemetry` since #41.

### Emission

* Counter `photon.resource.cache.access` with `photon.resource.cache.hit` and `photon.resource.type`, at all three exits of `getOrCreateResource (...)`:
  * cache disabled → **miss** (that is the point: a forgotten `setCacheEnabled (false)` in production then shows up as a 0 % hit ratio rather than as silence),
  * found in the read lock → hit,
  * not found → miss.
  Hits and misses are both counted, so the ratio is derivable.
* Counter `photon.resource.bundles.created`, incremented where `WebSiteResourceBundleManager.getResourceBundles (...)` sets `bCreatedAnyBundle = true`.
* Counter `photon.resource.bundles.skipped`, incremented in the `if (bResourcesAreOutOfSync)` branch of the startup read loop - one per persisted bundle that was discarded because a contained resource is missing or its hash changed.

### Deviation: no `type` attribute on the two bundle counters

The issue listed `photon.resource.type` for all three counters. It is on the cache counter, where `eResourceType` is a clean method parameter, but **not** on the two bundle counters: at both of those sites the resource type only exists per contained resource inside the loop, not per bundle, and hoisting it out would have meant either restructuring the loop or silently picking the first resource's type. Given that this is the lowest-priority issue of the set and that a bundle-type breakdown adds little, the counters carry no attribute. Easy to add later if it turns out to be wanted.

### Cardinality

`EWebSiteResourceType` is a two-value enum - safe. No resource **path** is used (bounded in principle, but easily in the hundreds and it multiplies with everything else; the log lines already carry it), and no content hash is used as an attribute at all. This is written down in the Javadoc of `ATTR_RESOURCE_TYPE`.

### No spans

None, as the issue decided: this is startup-time work that belongs to no request trace, and a parentless span adds noise without information. Recorded in the class Javadoc of `WebSiteResourceMetrics`.

The optional histogram over the content-hash computation in the `WebSiteResource` constructor was not added - the cache hit ratio answers the question this issue set out to answer, and the constructor cost is only paid on a miss, which the counter already exposes.

### Verification

Full reactor `mvn install` over all 23 modules: BUILD SUCCESS, no test failures, including the pre-existing `WebSiteResourceBundleManagerTest`. `README.md` news bullet added under `v10.6.0 - work in progress`.
